components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/StreamingListener.java

```diff
@@ -0,0 +1,163 @@
+            String pingInterval = natsProperties.getProperty(NatsConstants.NATS_STREAMING_PING_INTERVAL);
+            String traceConnection = natsProperties.getProperty(NatsConstants.NATS_STREAMING_TRACE_CONNECTION);
+
+            Options.Builder builder = new Options.Builder().natsUrl(StringUtils.isEmpty(natsStreamingUrl) ? NatsConstants.DEFAULT_NATS_STREAMING_URL : natsStreamingUrl)
+                    .clientId(natsStreamingClientId)
+                    .clusterId(StringUtils.isEmpty(natsStreamingClusterId) ? NatsConstants.DEFAULT_NATS_STREAMING_CLUSTER_ID : natsStreamingClusterId);
+
+            if (Boolean.parseBoolean(natsProperties.getProperty(NatsConstants.USE_CORE_NATS_CONNECTION))) {
+                builder.natsConn(new CoreListener(subject, injectHandler, natsProperties).getNatsConnection());
+            }
+
+            if (StringUtils.isNotEmpty(connectWait)) {
+                builder.connectWait(Duration.ofSeconds(Integer.parseInt(connectWait)));
+            }
+
+            if (StringUtils.isNotEmpty(discoverPrefix)) {
+                builder.discoverPrefix(discoverPrefix);
+            }
+
+            if (StringUtils.isNotEmpty(maxPingsOut)) {
+                builder.maxPingsOut(Integer.parseInt(maxPingsOut));
+            }
+
+            if (StringUtils.isNotEmpty(pingInterval)) {
+                builder.pingInterval(Duration.ofSeconds(Integer.parseInt(pingInterval)));
+            }
+
+            if (Boolean.parseBoolean(traceConnection)) {
+                builder.traceConnection();
+            }
+
+            StreamingConnectionFactory connectionFactory = new StreamingConnectionFactory(builder.build());
+            connection = connectionFactory.createConnection();
+        }
+        return true;
+    }
+
+    @Override public void consumeMessage(String sequenceName)
+            throws InterruptedException, IOException, TimeoutException {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        SubscriptionOptions.Builder subscriptionOptions = new SubscriptionOptions.Builder();
+
+        String durableName = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DURABLE_NAME);
+        String queueGroup = natsProperties.getProperty(NatsConstants.NATS_STREAMING_QUEUE_GROUP);
+        boolean isManualAck = Boolean.parseBoolean(natsProperties.getProperty(NatsConstants.NATS_STREAMING_MANUAL_ACK));
+        String ackWait = natsProperties.getProperty(NatsConstants.NATS_STREAMING_ACK_WAIT);
+        String maxInFlight = natsProperties.getProperty(NatsConstants.NATS_STREAMING_MAX_IN_FLIGHT);
+        String subscriptionTimeout = natsProperties.getProperty(NatsConstants.NATS_STREAMING_SUBSCRIPTION_TIMEOUT);
+        String dispatcher = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DISPATCHER);
+
+        if (StringUtils.isNotEmpty(durableName)) {
+            subscriptionOptions.durableName(durableName);
+        }
+
+        if (isManualAck) {
+            subscriptionOptions.manualAcks();
```

Comment:
We'll add a comment stating, if isManualAck = true, it will auto ack the message. If not, we need to call message.ack() for readability.